nestkernel/conn_builder.cpp

```diff
@@ -337,6 +337,61 @@ nest::ConnBuilder::single_connect_( index sgid,
   }
 }
 
+inline void
+nest::ConnBuilder::skip_conn_parameter_( thread target_thread, librandom::RngPtr& rng )
```

Comment:
I am wondering about the performance consequences of the parameter skipping as currently implemented. Just the looping and branching in this method will cost a bit, but I am particularly worried about randomized parameters. If I understand the code right, then every time we skip a target because it is not MPI- or thread-local, we call this method, and then obtain (and ignore) a value for each parameter defined. In some cases, this would mean drawing several random numbers, which may be quite demanding depending on the distribution used. Unless we can clearly demonstrate that this has no bad side effects even under "extreme" conditions (costly distributions, very many threads), we need to find a better solution.

One would be a flag "skipping required", which is set if there is at least one array parameter. One could then here just have `if ( not skipping_required_ ) return;`. The problem with this solution would be that different sequences of random numbers would be used for randomized parameters depending on the presence/absence of array parameters.

Therefore, I believe that a better approach would be to collect pointers to parameters requiring skipping and then skip only those. That would also avoid all the branching logic here, we could just do

```
for ( p = double_params_needing_skipping.begin() ; ... )
  p->double_value(...);
for ( p =int_params_needing_skipping.begin() ; ... )
  p->int_value(...);
```

One more issue: Can we be certain that the compiler will not optimize the value-calls away, since the value generated is never used? In that case, skipping would not work. To be safe, we need an explicit `skip()` method.